pkg/config/config.go

```diff
@@ -1445,9 +1450,80 @@ func load(config Config, origin string, loadSecret bool) (*Warnings, error) {
 	// setTracemallocEnabled *must* be called before setNumWorkers
 	warnings.TraceMallocEnabledWithPy2 = setTracemallocEnabled(config)
 	setNumWorkers(config)
+	setupFipsEndpoints(config)
 	return &warnings, nil
 }
 
+// setupFipsEndpoints overwrite the Agent endpoint for outgoing data to be sent to the local FIPS proxy. The local FIPS
+// proxy will be in charge of forwarding data to the Datadog backend following FIPS standard. Starting from
+// fips.port_range_start we will assign a dedicated port per product (metrics, logs, traces, ...).
+func setupFipsEndpoints(config Config) {
+	// Each port is dedicated to a specific data type:
+	//
+	// port_range_start: HAProxy stats
+	// port_range_start + 1:  metrics
+	// port_range_start + 2:  traces
+	// port_range_start + 3:  profiles
+	// port_range_start + 4:  processes
+	// port_range_start + 5:  logs
+	// port_range_start + 6:  databases monitoring metrics
+	// port_range_start + 7:  databases monitoring samples
+	// port_range_start + 8:  network devices metadata
+	// port_range_start + 9:  network devices snmp traps (unused)
+	// port_range_start + 10: instrumentation telemetry
+	// port_range_start + 11: appsec events (unused)
+
+	if !config.GetBool("fips.enabled") {
+		log.Debug("fips mode is disabled")
+		return
+	}
+
+	const (
+		proxyStats = iota
+		metrics
+		traces
+		profiles
+		processes
+		logs
+		databasesMonitoringMetrics
+		databasesMonitoringSamples
+		networkDevicesMetadata
+		networkDevicesSnmpTraps
+		instrumentationTelemetry
+		appsecEvents
+	)
+
+	localAddress := config.GetString("fips.local_address")
+	portRangeStart := config.GetInt("fips.port_range_start")
+	urlFor := func(port int) string { return fmt.Sprintf("%s:%d", localAddress, portRangeStart+port) }
+
+	log.Infof("Fips mode is enabled: all communication to DataDog Backend will be routed to the local FIPS proxy on '%s' starting from port %d", localAddress, portRangeStart)
+
+	// Metrics
+	config.Set("dd_url", "http://"+urlFor(metrics))
+
+	// APM
+	config.Set("apm_config.apm_dd_url", "http://"+urlFor(traces))
+	config.Set("apm_config.profiling_dd_url", "http://"+urlFor(profiles))
+	config.Set("apm_config.telemetry.dd_url", "http://"+urlFor(instrumentationTelemetry))
+
+	// Processes
+	config.Set("process_config.process_dd_url", "http://"+urlFor(processes))
+
+	// Logs
+	config.Set("logs_config.use_http", true)
+	config.Set("logs_config.logs_no_ssl", true)
+	config.Set("logs_config.logs_dd_url", urlFor(logs))
+
+	// Database monitoring
+	config.Set("database_monitoring.metrics.dd_url", urlFor(databasesMonitoringMetrics))
+	config.Set("database_monitoring.activity.dd_url", urlFor(databasesMonitoringMetrics))
+	config.Set("database_monitoring.samples.dd_url", urlFor(databasesMonitoringSamples))
+
+	// Network devices
+	config.Set("network_devices.metadata.dd_url", urlFor(networkDevicesMetadata))
```

Comment:
Should we unset `proxy.http` and `proxy.https` here ? I'm not sure if it has an impact when dd_url is set

Reply:
It does, very good point !